services/movie_session.py

```
@@ -0,0 +1,52 @@
+from datetime import datetime
+
+from django.db.models import QuerySet
+
+from db.models import MovieSession, Movie, CinemaHall
+
+
+def create_movie_session(
+        movie_show_time: datetime,
+        movie_id: int,
+        cinema_hall_id: int
+) -> MovieSession:
+    return MovieSession.objects.create(
+        show_time = movie_show_time,
+        movie_id = movie_id,
+        cinema_hall_id = cinema_hall_id
+    )
+
+
+def get_movies_sessions(session_date: str = None) -> QuerySet:
+    queryset = MovieSession.objects.all()
+    if session_date:
+        queryset = queryset.filter(show_time__date=session_date)#!!!
+
+    return queryset
+
+
+def get_movie_session_by_id(
+        movie_session_id: int
+) -> MovieSession:
+    return MovieSession.objects.get(id=movie_session_id)
+
+
+def update_movie_session(
+        session_id: int,
+        show_time: str = None,
+        movie_id: int = None,
+        cinema_hall_id: int = None
+) -> None:
+    movie_session = MovieSession.objects.get(id = session_id)
```

The `update_movie_session` function uses `MovieSession.objects.get(id=session_id)`, which will raise a `DoesNotExist` exception if the session ID is not found. Consider handling this exception to avoid server errors.